File: replace_theme.py
import os
import re

files = [
    r"c:\Users\shlokui\Desktop\CollegeProject\index.html",
    r"c:\Users\shlokui\Desktop\CollegeProject\css\styles.css",
    r"c:\Users\shlokui\Desktop\CollegeProject\js\app.js"
]

replacements = {
    r"#111417": "#2D4159",
    r"#1a1d21": "#59253A",
    r"#2a2d32": "#78244C",
    r"#3d4148": "#895061",
    r"#CCFF00": "#0677A1",
    r"#b8e600": "#0677A1",
    r"#e0ff66": "#78244C",
    r"#00ff88": "#78244C",
    r"rgba\(\s*204\s*,\s*255\s*,\s*0\s*,": "rgba(6, 119, 161,",
    r"rgba\(\s*204\s*,\s*255\s*,\s*0\s*\)": "rgba(6, 119, 161)",
    r"rgba\(\s*0\s*,\s*255\s*,\s*136\s*,": "rgba(120, 36, 76,",
    r"rgba\(\s*0\s*,\s*255\s*,\s*136\s*\)": "rgba(120, 36, 76)",
}

for file_path in files:
    if not os.path.exists(file_path):
        continue
    
    with open(file_path, "r", encoding="utf-8") as f:
        content = f.read()

    for pattern, replacement in replacements.items():
        content = re.sub(pattern, replacement, content, flags=re.IGNORECASE)

    with open(file_path, "w", encoding="utf-8") as f:
        f.write(content)

print("Theme colors updated successfully!")

File: replace_theme.js
const fs = require('fs');

const files = [
    'c:/Users/shlokui/Desktop/CollegeProject/index.html',
    'c:/Users/shlokui/Desktop/CollegeProject/css/styles.css',
    'c:/Users/shlokui/Desktop/CollegeProject/js/app.js'
];

const replacements = [
    { pattern: /#111417/gi, replacement: '#2D4159' },
    { pattern: /#1a1d21/gi, replacement: '#59253A' },
    { pattern: /#2a2d32/gi, replacement: '#78244C' },
    { pattern: /#3d4148/gi, replacement: '#895061' },
    { pattern: /#CCFF00/gi, replacement: '#0677A1' },
    { pattern: /#b8e600/gi, replacement: '#0677A1' },
    { pattern: /#e0ff66/gi, replacement: '#78244C' },
    { pattern: /#00ff88/gi, replacement: '#78244C' },
    { pattern: /rgba\(\s*204\s*,\s*255\s*,\s*0\s*,/gi, replacement: 'rgba(6, 119, 161,' },
    { pattern: /rgba\(\s*204\s*,\s*255\s*,\s*0\s*\)/gi, replacement: 'rgba(6, 119, 161)' },
    { pattern: /rgba\(\s*0\s*,\s*255\s*,\s*136\s*,/gi, replacement: 'rgba(120, 36, 76,' },
    { pattern: /rgba\(\s*0\s*,\s*255\s*,\s*136\s*\)/gi, replacement: 'rgba(120, 36, 76)' },
];

files.forEach(file => {
    if (fs.existsSync(file)) {
        let content = fs.readFileSync(file, 'utf8');
        replacements.forEach(r => {
            content = content.replace(r.pattern, r.replacement);
        });
        fs.writeFileSync(file, content, 'utf8');
        console.log(`Updated ${file}`);
    }
});
